### ISSUE TITLE
GE multi-echo not right...

### ISSUE THREAD

**Issue by andersonwinkler** [rordenlab/dcm2niix#209]
Hi Chris,

Thanks for this wonderful tool (like all the others you've made!).

We have some multi-echo EPIs acquired on a GE scanner (4 echoes, 196 volumes per echo, 784 total vols). After sorting the DICOM files into one directory per echo, then running dcm2niix 4 times, we'd expect to find 1 4D NIFTI per echo, each with 196 volumes.

However, we are getting instead, for each echo, 1 4D NIFTI with 196/4 = 49 volumes. Each volume contains then 4 brains stacked on each other along the z-direction (i.e., concatenated on space).

Below a screenshot. On the left what we expect. On the right what we get.

(Btw, if we don't sort the DICOMs into 4 separate directories, the reconstruction gets all wrong).

Any idea?

Thanks!

Anderson

<img width="1385" alt="screen shot 2018-07-14 at 20 45 47" src="https://user-images.githubusercontent.com/1075764/42729513-a3e33308-87a8-11e8-89b9-920469f604f3.png">

**Comment by neurolabusc** [comment]
I have updated the software, so it now gives precedence to GE private tag 0021,104F for determining number of slices in GE datasets. This will correctly convert your images once you have segmented them into different folders using your Python script.

In theory, we could update dcm2niix to mimic your Python script, detecting that tag 0019,10a2 varies for the different echos even though the reported echo time, instance number, slice number, etc are repeated. This is complicated by the fact that we 0019,10a2 is not indexed from a fixed value, and is repeated for all the timepoints of a given slice at a specific echo time. Even if we did resolve this, the recorded echo times would be incorrect, as the DICOMs report only a single echo despite reality. Therefore, the current dcm2niix reports that there are duplicates and the user can use a script to rectify this while the vendor fixes this issue.
